src/effects/effectchainslot.cpp

```diff
@@ -192,13 +192,10 @@ void EffectChainSlot::loadEffect(const unsigned int iEffectSlotNumber,
                                                  m_enabledInputChannels);
 }
 
-void EffectChainSlot::refreshAllEffects() {
+void EffectChainSlot::reloadAllEffects() {
     for (EffectSlotPointer pSlot : m_effectSlots) {
-        if (pSlot->isLoaded()) {
-            pSlot->loadEffect(pSlot->getManifest(), pSlot->getInstantiator(), m_enabledInputChannels);
-        }
+        pSlot->reload();
```

Comment:
Keep passing `m_enabledInputChannels` here. It was only the EffectManifest and EffectInstantiator arguments that should have been removed. This keeps EffectSlot's state in EffectSlot only and EffectChainSlot in EffectChainSlot only.

Reply:
Requiring that argument should ensure that `EffectSlot::reload` is only ever called by EffectChainSlot. There will be a similar situation when implementing EffectPreset loading.